### PR TITLE
Improve Minio operator

### PR DESCRIPTION
**Pull request by galexrt**
**Description of your changes:**
* Renamed MinioController to Controller as the package is named minio
* Update Minio Object Store objects if they already exist.
* Added readiness and liveness probes to Minio StatefulSet.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

File: pkg/operator/minio/controller.go [hunk 1]
```diff
@@ -103,13 +103,22 @@ func (c *MinioController) makeMinioHeadlessService(name, namespace string, spec
...
 			ClusterIP: v1.ClusterIPNone,
 		},
-	})
+	}
 	k8sutil.SetOwnerRef(c.context.Clientset, namespace, &svc.ObjectMeta, &ownerRef)
 
-	return svc, err
+	svc, err := c.context.Clientset.CoreV1().Services(namespace).Create(svc)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return nil, err
```
Comment (travisn) [inline pkg/operator/minio/controller.go]: Do we need to log, or return a composite error such as this? It's been helpful to debug the source of errors with that pattern.
`return nil, fmt.Errorf("failed to create service. %+v", err)`

File: pkg/operator/minio/controller.go [hunk 2]
```diff
@@ -207,14 +242,13 @@ func validateObjectStoreSpec(spec miniov1alpha1.ObjectStoreSpec) error {
...
-func (c *MinioController) makeMinioStatefulSet(name, namespace string, spec miniov1alpha1.ObjectStoreSpec, ownerRef meta_v1.OwnerReference) (*v1beta2.StatefulSet, error) {
-	appsClient := c.context.Clientset.AppsV1beta2()
-
+func (c *Controller) makeMinioStatefulSet(name, namespace string, spec miniov1alpha1.ObjectStoreSpec, ownerRef meta_v1.OwnerReference) (*v1beta2.StatefulSet, error) {
 	accessKey, secretKey, err := c.getAccessCredentials(spec.Credentials.Name, spec.Credentials.Namespace)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO Use a Secret object (`envFrom`) instead of putting the keys in plain text in the StatefulSet
```
Comment (travisn) [inline pkg/operator/minio/controller.go]: instead of a TODO, is there an issue tracking this work item?

Comment (galexrt) [reply pkg/operator/minio/controller.go]: Created: https://github.com/rook/rook/issues/2465

File: pkg/operator/minio/controller.go (hunk 1 above)
Comment (galexrt) [reply pkg/operator/minio/controller.go]: I have gone ahead and made the `make...()` funcs return full errors and the using part to then log them out.

**Comment by galexrt** [comment]
@jbw976 If I get the time I'll probably take a look at that too.

**Pull request merged**